src/main/java/sirius/biz/tenants/TenantUserManager.java

```diff
@@ -293,6 +300,65 @@ public UserInfo findUserByUserId(String accountId) {
         return asUser(account, null);
     }
 
+    @Nonnull
+    @Override
+    public UserInfo bindToRequest(@Nonnull WebContext ctx) {
+        UserInfo info = super.bindToRequest(ctx);
+        return verifyIpRange(ctx, info);
+    }
+
+    /**
+     * Checks if the current user violates the configured ip range constraints.
+     * <p>
+     * Will remove permissions if the users ip address does not match the configured
+     * ip range.
+     * <p>
+     * This is a security feature to prevent unwanted access to certain features from anywhere but a valid ip range.
+     *
+     * @param ctx  the current request
+     * @param info the current user
+     * @return the current user but possibly with less permissions
+     */
+    private UserInfo verifyIpRange(WebContext ctx, UserInfo info) {
+        String actualUser = ctx.getSessionValue("default-user-id").asString();
+
+        UserInfo realUser = findUserByUserId(actualUser);
```

Comment:
why not fetchAccount(..).getTenant().getValue().... with appropriate null checks

Reply:
this should all be served from the cache

Reply:
findUserByUserId calls fetchAccount(accountId, null).
tryAs should use the cached tenant if the tenant already exists within the cache